packages/firestore/src/local/shared_client_state.ts

```diff
@@ -674,6 +674,15 @@ export class WebStorageSharedClientState implements SharedClientState {
     error?: FirestoreError
   ): void {
     this.persistMutationState(batchId, state, error);
+
+    // Once a final mutation result is observed by other clients, they no longer
+    // access the mutation's metadata entry. Since LocalStorage replays events
+    // in order, it is safe to delete the entry right after updating it.
+    // We enqueue the delete on the queue since LocalStorage writes acquire
+    // a lock internally, and we don't need to wait for the delete to succeed.
```

Comment:
So we're enqueuing the delete as a performance optimization?  Do we have reason to believe this is going to make a meaningful difference in some scenario?  Else I'd keep the code simple for now.

(FWIW, persistMutationState already is writing to LocalStorage, acquiring the lock, etc. so it's not like this will be any more expensive)

Reply:
Optimize first, remove later. Ok, maybe not... I removed the only L6-worthy part of this PR.